### PR TITLE
[Python/Tornado] Refactor with Pypy

### PR DESCRIPTION
**Pull request by Dith3r**

### CODE REVIEW

File: frameworks/Python/tornado/benchmark_config.json [hunk 1]
```diff
@@ -19,14 +21,40 @@
       "webserver": "Tornado",
       "os": "Linux",
       "database_os": "Linux",
-      "display_name": "Tornado-raw",
+      "display_name": "Tornado-Py2-Mongo",
```
Comment (msmith-techempower) [inline frameworks/Python/tornado/benchmark_config.json]: Please change this to "Tornado" (if it is not a raw database test any longer).

The `py2` and `mongo` metadata is captured and displayed elsewhere.

**Comment by methane** [comment]
http://docs.python.org/2.7/howto/sorting.html

Use keyfunc insted of cmp func.

File: frameworks/Python/tornado/benchmark_config.json [hunk 2]
```diff
@@ -61,9 +91,9 @@
       "webserver": "Tornado",
       "os": "Linux",
       "database_os": "Linux",
-      "display_name": "Tornado-raw",
+      "display_name": "Tornado",
       "notes": "CPython 3.4",
```
Comment (methane) [inline frameworks/Python/tornado/benchmark_config.json]: Please clear this note.

**Comment by methane** [comment]
@Dith3r Could you merge _py2 and _py3 version?  How difficult?

File: frameworks/Python/tornado/server_py3.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            world = await world
+            new_value = randint(1, 10000)
+            await db.World.update_one({'_id': world['_id']}, {"$set": {self.RANDOM_NUMBER: new_value}})
+
+            worlds.append({self.ID: int(world['_id']),
+                    self.RANDOM_NUMBER: world[self.RANDOM_NUMBER]})
+        self.finish(json.dumps(worlds))
+
+
+class FortuneHandler(JsonHandler):
```
Comment (methane) [inline frameworks/Python/tornado/server_py3.py]: BaseHandler?

Comment (Dith3r) [inline frameworks/Python/tornado/server_py3.py]: JsonHandler add headers for content type and it inheritance from base handler.

**Comment by Dith3r** [comment]
Python2 version is different from Python3 version. In Python3 I used asycio ioloop and specific syntax only available in this version (async def, async for, await, etc.). This checks too different approaches with tornado.

File: frameworks/Python/tornado/server_py3.py (hunk 3 above)
Comment (methane) [inline frameworks/Python/tornado/server_py3.py]: Doesn't FortuneHandler returns HTML?

**Comment by methane** [comment]
OK, LGTM

**Comment by nbrady-techempower** [comment]
Not passing Travis

File: frameworks/Python/tornado/templates/fortunes.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{% whitespace oneline %}
+<!DOCTYPE html>
+<html>
+<head><title>Fortunes</title></head>
+<body>
+<table>
+<tr><th>id</th><th>message</th></tr>
+{% for fortune in fortunes %}
+<tr><td>{{ fortune['id'] }}</td><td>{{ fortune['message'] }}</td></tr>
```
Comment (methane) [inline frameworks/Python/tornado/templates/fortunes.html]: `{{ escape(fortune['message']) }}`

**Comment by nbrady-techempower** [comment]
@Dith3r Can this be fixed?

**Comment by Dith3r** [comment]
I will provide fix later today.

File: frameworks/Python/tornado/server_pg.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import tornado.options
+from tornado.options import options
+import tornado.httpserver
+from commons import JsonHandler, JsonHelloWorldHandler, PlaintextHelloWorldHandler, BaseHandler
+
+
+tornado.options.define('port', default=8888, type=int, help="Server port")
+tornado.options.define('postgres', default=None,
+                       type=str, help="PostgreSQL host")
+tornado.options.define('backlog', defualt=8192, type=int,
```
Comment (methane) [inline frameworks/Python/tornado/server_pg.py]: default

**Comment by Dith3r** [comment]
Someone could trigger check?

**Comment by msmith-techempower** [comment]
No idea why this kicked off a full Travis-CI run, but the Python tests pass; LGTM!

**Pull request merged**